CONTRIBUTING.md

```diff
@@ -0,0 +1,281 @@
+
+Here are some high-level tips on writing a good tutorial:
+
+- Focus on guiding the user to get something done, but keep in mind the end-goal is more to impart principles than to create a perfect production system
+- Be specific, not abstract and follow one path
+  - No need to go deeply into alternative approaches, but it’s ok to reference them, ideally with a link to an appropriate how-to guide
+- Get "a point on the board" as soon as possible - something the user can run that outputs something
+  - You can iterate and expand afterwards
+  - Try to frequently checkpoint at given steps where the user can run code and see progress
+- Focus on results, not technical explanation
+  - Crosslink heavily to appropriate conceptual/reference pages
+- The first time you mention a LangGraph concept, use its full name (e.g. "human-in-the-loop"), and link to its conceptual/other documentation page
+  - It's also helpful to add a prerequisite callout that links to any pages with necessary background information
+- End with a recap/next steps section summarizing what the tutorial covered and future reading, such as related how-to guides
+
+### How-to guides
+
+A how-to guide, as the name implies, demonstrates how to do something discrete and specific.
+It should assume that the user is already familiar with underlying concepts, and is trying to solve an immediate problem, but
+should still give some background or list the scenarios where the information contained within can be relevant.
+They can and should discuss alternatives if one approach may be better than another in certain cases.
+
+To quote the Diataxis website:
+
+> A how-to guide serves the work of the already-competent user, whom you can assume to know what they want to do, and to be able to follow your instructions correctly.
+
+Some examples include:
+
+- [How to add persistence to your graph](https://langchain-ai.github.io/langgraph/how-tos/persistence/)
+- [How to view and update past graph state](https://langchain-ai.github.io/langgraph/how-tos/human_in_the_loop/time-travel/)
+
+Here are some high-level tips on writing a good how-to guide:
+
+- Clearly explain what you are guiding the user through at the start
+- Assume higher intent than a tutorial and show what the user needs to do to get that task done
+- Assume familiarity of concepts, but explain why suggested actions are helpful
+  - Crosslink heavily to conceptual/reference pages
+- Discuss alternatives and responses to real-world tradeoffs that may arise when solving a problem
+- Use lots of example code
+- End with a recap/next steps section summarizing what the tutorial covered and future reading, such as other related how-to guides
+
+### Conceptual guide
+
+LangGraph's conceptual guide falls under the **Explanation** quadrant of Diataxis. They should cover LangChain terms and concepts
+in a more abstract way than how-to guides or tutorials, and should be geared towards curious users interested in
```

Comment:
Hm - IDK if I love recommending against code examples. I get that it's a decent classification feature "if there's code, it's probably too specific", but if I want to go deep in learning about a software library, I prefer to have explanations in code.

Another thing conceptual guide answers is **why** things work they way they do and **why** we designed things a certain way. Or **what if** things worked differently, etc.

Reply:
yea, i think that's fair (lifted this from langchain docs, so we might want to update that as well) cc @jacoblee93

Reply:
Yeah down to modify - sentiment is "don't use excessive code blocks" since that trends more towards how-tos. A few is fine